helper-cli/src/main/kotlin/commands/ConvertToDependencyGraphCommand.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands
+
+import com.fasterxml.jackson.databind.JsonMappingException
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.ProgramResult
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.analyzer.AnalyzerResultBuilder
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.model.utils.DependencyGraphConverter
+import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.utils.expandTilde
+
+/**
+ * A command to convert an ORT result file from the classic dependency tree format to the dependency graph format.
+ *
+ * The input file can contain the serialized form of either a full [OrtResult], an [AnalyzerResult], or a single
+ * [ProjectAnalyzerResult].
+ */
+class ConvertToDependencyGraphCommand : CliktCommand(
+    help = "Convert an ORT result from the dependency tree representation to the dependency graph format."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input. This can be the serialized form of an OrtResult, an " +
+                "AnalyzerResult, or a ProjectAnalyzerResult."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val outputFile by option(
+        "--output-file", "-o",
+        help = "The file in which to write the result of the conversion. If missing, the input file is overwritten."
```

Comment:
Is "If missing, the input file is overwritten." really needed? Why did you decide to add this?

Reply:
The tool is going to be used to convert the files with the test expectations to the new format. I thought, for this use case it is more convenient to do the modification in-place, rather than write the result to a temporary file, delete the original file, and eventually rename the output file.